lib/chat_web/live/main_live/page/lobby.ex

```diff
@@ -87,26 +85,52 @@ defmodule ChatWeb.MainLive.Page.Lobby do
     room = Rooms.add_request(room_hash, me, time)
     Log.request_room_key(me, time, room.pub_key)
 
-    ChangeTracker.on_saved(fn ->
+    # ChangeTracker.on_saved(fn ->
+    #  PubSub.broadcast!(
+    #    Chat.PubSub,
+    #    @topic,
+    #    {:room_request, room_hash, me |> Identity.pub_key() |> Utils.hash()}
+    #  )
+    # end)
+
+    socket
+    |> assign_room_list()
+  end
+
+  def approve_room_request(%{assigns: %{room_map: room_map}} = socket, room_hash, user_hash) do
+    if room_map[room_hash] do
+      {_, _, encrypted_room_identity} =
+        room_hash
+        |> Rooms.approve_request(user_hash, room_map[room_hash])
+        |> Rooms.get_request(user_hash)
+
       PubSub.broadcast!(
         Chat.PubSub,
         @topic,
-        :room_request
+        {:room_request_approved, encrypted_room_identity, user_hash}
       )
-    end)
+    end
 
     socket
-    |> assign_room_list()
   end
 
-  def approve_requests(socket) do
-    socket
-    |> approve_joined_room_requests()
-  end
+  def join_approved_room(
+        %{assigns: %{me: me, my_id: my_id, monotonic_offset: time_offset}} = socket,
+        enctypted_room_entity,
+        user_hash
+      ) do
+    if my_id == user_hash do
+      new_room_identity = Rooms.decrypt_room_identity(enctypted_room_entity, me)
+      room_hash = new_room_identity |> Identity.pub_key() |> Utils.hash()
+      time = Chat.Time.monotonic_to_unix(time_offset)
+      Rooms.join_approved_requests(room_hash, me, time)
```

Comment:
DB might not have request approved or even created. It may be in queue.
We need a way for room to forget about request disregarding its state